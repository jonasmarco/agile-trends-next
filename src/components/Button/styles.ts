import styled, { css, DefaultTheme } from 'styled-components'
import { ButtonProps } from '.'

type WrapperProps = {} & Pick<ButtonProps, 'fullWidth'>

const wrapperSizeModifiers = {
  fullWidth: () => css`
    width: 100%;
  `
}

export const Wrapper = styled.button<WrapperProps>`
  ${({ theme, fullWidth }) => css`
    cursor: pointer;
    height: 4rem;
    padding: 0 4.4rem;

    ${!!fullWidth && wrapperSizeModifiers.fullWidth()};
  `}
`
