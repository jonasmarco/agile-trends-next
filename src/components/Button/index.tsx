import * as S from './styles'

export type ButtonProps = {
  children?: React.ReactNode
  fullWidth?: boolean
  onClick?: () => (event: React.MouseEvent<HTMLButtonElement>) => void
}

const Button = ({ children, fullWidth = false, ...props }: ButtonProps) => (
  <S.Wrapper fullWidth={fullWidth} {...props}>
    {!!children && <span>{children}</span>}
  </S.Wrapper>
)

export default Button
