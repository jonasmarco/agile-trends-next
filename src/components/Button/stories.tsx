import { Story, Meta } from '@storybook/react/types-6-0'
import Button from '.'

export default {
  title: 'Button',
  component: Button,
  argTypes: {
    children: {
      type: 'string'
    },
    fullWidth: {
      type: 'boolean'
    }
  }
} as Meta

export const Basic: Story = (args) => <Button {...args} />

Basic.args = {
  children: 'Simular agora'
}
