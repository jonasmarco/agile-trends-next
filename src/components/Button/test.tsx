import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/tests/helpers'

import Button from '.'

describe('<Button />', () => {
  it('should render the component correctly', () => {
    const { container } = renderWithTheme(<Button>Simular agora</Button>)

    expect(screen.getByRole('button', { name: /Simular agora/i })).toHaveStyle({
      height: '4rem'
    })

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render a fullWidth version', () => {
    renderWithTheme(<Button fullWidth>Simular agora</Button>)

    expect(screen.getByRole('button', { name: /Simular agora/i })).toHaveStyle({
      width: '100%'
    })
  })
})
