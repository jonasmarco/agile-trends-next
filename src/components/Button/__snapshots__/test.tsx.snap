// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render the component correctly 1`] = `
.c0 {
  cursor: pointer;
  height: 4rem;
  padding: 0 4.4rem;
}

<button
  class="c0"
>
  <span>
    Simular agora
  </span>
</button>
`;
